adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -587,8 +587,10 @@ void onActivityResult(final int requestCode, final int resultCode, final Intent
                     final String tenantId = data.getStringExtra(
                             AuthenticationConstants.Broker.ACCOUNT_USERINFO_TENANTID);
                     final UserInfo userinfo = UserInfo.getUserInfoFromBrokerResult(data.getExtras());
+                    final String authority = data.getStringExtra(AuthenticationConstants.Broker.AUTHORITY);
                     final AuthenticationResult brokerResult = new AuthenticationResult(accessToken, null,
                             expire, false, userinfo, tenantId, idtoken, null);
+                    brokerResult.setCloudInstanceName(authority);
```

Comment:
if this is the complete authority  url with host and tenant name in it then broker result return Authority and not CloudInstanceName. they are 2 different things and lack of clarity can create a bug pit.